### ISSUE TITLE
Training Demucs on Google Colab

### ISSUE THREAD

**Issue by MaxC2** [facebookresearch/demucs#137]
Thank you very much for this great project, work very well on Google Colab for me. 

Unfortunately, I don't have a modern computer nor a modern GPU locally, so my question is both :

1) Is the training possible on Google Colab free or not?
2) What is the training duration time for the MusDB for the pretrained models with his config (8 x V100 with 32GB of RAM) ?

**Comment by adefossez** [comment]
Hey,
It is possible to train on Colab free, but you will have to use the smaller model (`--channels=64`) because of the ram.
It takes about 12 hours with 8 V100 to train. So I would expect it to train in around a week on Colab, the most painful aspect being to keep the notebook active during that time

**Comment by MaxC2** [comment]
Yes, Google Colab Pro seem to no be available in others country than US and Canada. More, the paid version is not a guaranty of stability, you can be kicked out too within 24 hours. 

I'm very newbie in AI training consideration. As I understand, maybe the idea is to split the training in chunck of checkpoints and resume the process day by day. Do you think it's possible?

**Comment by adefossez** [comment]
Yes, you can mount google drive for instance and store the checkpoints there (using the `--checkpoints` option).
Then also set `-r 1 -e 240` (I defined one epoch as 2 passes over the dataset, this will make it only one and checkpoint more often which for colab is better as there will be a lot of interruption).
The process will be a bit painful but probably doable.
You should also store the musdb dataset on google drive.

**Comment by burntscarr** [comment]
I’ll just leave this here
I have Colab Pro for the next 25 days if you’d like me to test something? I leave my pc on and can automate a mouse movement or something to keep the kill switch from activating.
